src/@hooks/useModalStore.ts

```diff
@@ -61,4 +87,22 @@ export const useModalStore = create<ModalState>((set, get) => ({
 
 export const useModalState = () => useModalStore();
 
+export const useConfirmModalState = () =>
+  useModalStore(
+    state => ({
+      message: state.confirmMessage,
+      isOpenConfirmModal: state.isOpenConfirmModal,
+      onConfirm: state.onConfirm,
+    }),
+    shallow,
+  );
+
+export const useSetConfirmModal = () =>
+  useModalStore(state => {
+    return {
+      openConfirm: state.actions.openConfirm,
+      closeConfirm: state.actions.closeConfirm,
+    };
+  }, shallow);
```

Comment:
```suggestion
  useModalStore(state => ({
    openConfirm: state.actions.openConfirm,
    closeConfirm: state.actions.closeConfirm,
  }), shallow);
```

위에 useConfirmModalState 함수와 코드랑 형식이 맞으면 좋겠네요~